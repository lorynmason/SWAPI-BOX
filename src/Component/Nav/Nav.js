import React from 'react'
import PropTypes from 'prop-types'

const Nav = ( {favorites, changePage} ) => {

  return (
    <nav className="nav">
      <button className="nav-link" onClick={() => changePage('characters')} >Characters</button>
      <button className="nav-link" onClick={() => changePage('planets')}>Planets</button>
      <button className="nav-link" onClick={() => changePage('vehicles')}>Vehicles</button>
      <button className="nav-link" onClick={() => changePage('favorites')}>Favorites: {favorites.length}</button>
    </nav>
  )
}

Nav.propTypes = {
  favorites: PropTypes.array.isRequired,
  changePage: PropTypes.func.isRequired
}

export default Nav