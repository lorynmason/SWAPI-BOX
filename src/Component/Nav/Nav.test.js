import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

describe('Card', () => {
  let wrapper
  let mockFavorites = [{
    id: 'R2-D2'
  }]
  const mockFunc = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<Nav favorites={mockFavorites} changePage={mockFunc}/>)
  })

  it('should match snapshot',() => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call changePage when clicked',() => {
    wrapper.find('#characters-link').simulate('click');
    wrapper.find('#planets-link').simulate('click');
    wrapper.find('#vehicles-link').simulate('click');
    wrapper.find('#favorites-link').simulate('click');
    expect(mockFunc).toHaveBeenCalledTimes(4)
  })
})