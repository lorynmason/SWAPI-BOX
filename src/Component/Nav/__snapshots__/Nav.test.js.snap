// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    changePage={[MockFunction]}
    favorites={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
        >
          Characters
        </button>,
        <button
          onClick={[Function]}
        >
          Planets
        </button>,
        <button
          onClick={[Function]}
        >
          Vehicles
        </button>,
        <button>
          Favorites: 
          0
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Characters",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Characters",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Favorites: ",
          0,
        ],
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
          >
            Characters
          </button>,
          <button
            onClick={[Function]}
          >
            Planets
          </button>,
          <button
            onClick={[Function]}
          >
            Vehicles
          </button>,
          <button>
            Favorites: 
            0
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Characters",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Characters",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Favorites: ",
            0,
          ],
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
